numpy/core/numeric.py

```diff
@@ -401,11 +402,17 @@ def asarray(a, dtype=None, order=None):
         Input data, in any form that can be converted to an array.  This
         includes lists, lists of tuples, tuples, tuples of tuples, tuples
         of lists and ndarrays.
-    dtype : data-type, optional
+    dtype : {None, data-type}, optional
         By default, the data-type is inferred from the input data.
-    order : {'C', 'F'}, optional
-        Whether to use row-major ('C') or column-major ('F' for FORTRAN)
-        memory representation.  Defaults to 'C'.
+    order : {'C', 'F', 'A'}, optional
+        Whether to use row-major ('C'), column-major ('F'), or any ('A')
+        memory representation.  Defaults to 'A'.
+    precision: {None, dtype, ndarray, scalar}, optional
```

Comment:
I'm following the `result_type` arguments. The `0.0` version behaves differently than specifying a type, in that ints are promoted to double, but floats retain their type (arr + scalar same kind rules). I don't know if that really matters here, but one of the uses of this PR is to replace the `arr +  0.0` constructions. Using float32 instead, smaller integer types are converted to float32. This may be OK, as in numpy most lapack routines are double only, and float64 is probably a good alternative.